### PR TITLE
refactor: refactor GreptimeDB k8s deployment part

### PR DESCRIPTION
**Pull request by WenyXu**
## What's Changed in this PR

refactor GreptimeDB k8s deployment part

## Checklist

- [ ] Please confirm that all corresponding versions of the documents have been revised.
- [ ] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 1]
```diff
@@ -1,44 +1,105 @@
...
-make sure you have already installed [GreptimeDB Operator](./manage-greptimedb-operator/deploy-greptimedb-operator.md) on your Kubernetes cluster.
+This guide will walk you through deploying a GreptimeDB cluster on a Kubernetes (K8s) environment. Make sure the [GreptimeDB Operator](./manage-greptimedb-operator/deploy-greptimedb-operator.md) is installed on your cluster before proceeding. We’ll cover everything from setting up an etcd cluster (optional) to configuring options and connecting to the database.
 
-## Using Helm for deployment
+## Prerequisites
+- [Helm](https://helm.sh/docs/intro/install/) (Use the Version appropriate for your Kubernetes API version)
+- [GreptimeDB Operator](./manage-greptimedb-operator/deploy-greptimedb-operator.md) (Assumes the local host has a matching installation of the GreptimeDB Operator) 
 
-### Create an etcd cluster
+## Create an etcd cluster (Optional)
```
Comment (nicecui) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:9]: Need to explain why this section is optional

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 2]
```diff
@@ -1,44 +1,105 @@
...
+```bash
+NAME                                      READY   STATUS    RESTARTS   AGE
+my-greptimedb-datanode-0                  1/1     Running   0          30s
+my-greptimedb-datanode-2                  1/1     Running   0          30s
+my-greptimedb-datanode-1                  1/1     Running   0          30s
+my-greptimedb-frontend-7476dcf45f-tw7mx   1/1     Running   0          16s
+my-greptimedb-meta-689cb867cd-cwsr5       1/1     Running   0          31s
+```
+
+## Advanced Configuration Options
```
Comment (nicecui) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:67]: Maybe a link to the configuration.md file, and explain the rules for transforming the toml options into helm command options?

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 3]
```diff
@@ -1,44 +1,105 @@
...
-First, establish an etcd cluster to support GreptimeDB by executing the following command:
+To install the ectd cluster, run the following `helm install` command. This command also creates a dedicated namespace `etcd-cluster` for the installation.
 
 ```shell
-helm upgrade \
-  --install etcd oci://registry-1.docker.io/bitnamicharts/etcd \
+helm install etcd oci://registry-1.docker.io/bitnamicharts/etcd \
   --set replicaCount=3 \
-  --set auth.rbac.create=false \
+  --set auth.rbac.create=false \****
```
Comment (nicecui) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:16]: what does the `****` mean?

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 4]
```diff
@@ -53,47 +114,48 @@ helm install greptimedb greptime/greptimedb-cluster \
   -n greptimedb-cluster
 ```
 
-For a comprehensive list of configurable values via Helm,
-please refer to the [value configuration](https://github.com/GreptimeTeam/helm-charts/blob/main/charts/greptimedb-cluster/README.md#values).
+### Complex Configurations with values.yaml
```
Comment (nicecui) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:117]:
```suggestion
### Complex configurations with values.yaml
```

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 5]
```diff
@@ -53,47 +114,48 @@ helm install greptimedb greptime/greptimedb-cluster \
...
 
-<!-- ### Use a different GreptimeDB version
+For more intricate configurations, first download the default values.yaml file and modify it locally.
```
Comment (nicecui) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:119]:
```suggestion
For more intricate configurations, first download the default `values.yaml` file and modify it locally.
```

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 6]
```diff
@@ -53,47 +114,48 @@ helm install greptimedb greptime/greptimedb-cluster \
...
-  frontend:
-    replicas: 1
-  meta:
-    replicas: 1
-    etcdEndpoints:
-      - "etcd.etcd-cluster.svc.cluster.local:2379"
-  datanode:
-    replicas: 3
+Edit the file to include your custom settings. Here’s an example:
+
```
Comment (nicecui) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:126]: could you please add the authentication example code?

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 7]
```diff
@@ -1,44 +1,105 @@
 # Deploy GreptimeDB Cluster
 
-Before the deployment,
-make sure you have already installed [GreptimeDB Operator](./manage-greptimedb-operator/deploy-greptimedb-operator.md) on your Kubernetes cluster.
+This guide will walk you through deploying a GreptimeDB cluster on a Kubernetes (K8s) environment. Make sure the [GreptimeDB Operator](./manage-greptimedb-operator/deploy-greptimedb-operator.md) is installed on your cluster before proceeding. We’ll cover everything from setting up an etcd cluster (optional) to configuring options and connecting to the database.
```
Comment (daviderli614) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:3]:
```suggestion
This guide will walk you through deploying a GreptimeDB cluster on a Kubernetes(K8S) environment. Make sure the [GreptimeDB Operator](./manage-greptimedb-operator/deploy-greptimedb-operator.md) is installed on your cluster before proceeding. We’ll cover everything from setting up an etcd cluster (optional) to configuring options and connecting to the database.
```

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 8]
```diff
@@ -1,44 +1,105 @@
...
+Check that all pods are running properly in the `greptimedb-cluster` namespace:
+
+```bash
+kubectl get pods -n greptimedb-cluster
+```
+
+You should see output similar to this:
+```bash
+NAME                                      READY   STATUS    RESTARTS   AGE
+my-greptimedb-datanode-0                  1/1     Running   0          30s
```
Comment (daviderli614) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:59]: Why is there a prefix of "my"?
Your installation command is this:
```
helm install greptimedb greptime/greptimedb-cluster \
  --set meta.etcdEndpoints=etcd.etcd-cluster.svc.cluster.local:2379 \
  --create-namespace \
  -n greptimedb-cluster
```

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 9]
```diff
@@ -1,44 +1,105 @@
...
+```bash
+helm install \
+  --set meta.etcdEndpoints=etcd.etcd-cluster.svc.cluster.local:2379 \
+  --set meta.enableRegionFailover=true \ 
+  --set objectStorage.s3.bucket="your-bucket" \
+  --set objectStorage.s3.region="region-of-bucket" \
+  --set objectStorage.s3.root="root-directory-of-data" \
+  --set objectStorage.credentials.accessKeyId="your-access-key-id" \
+  --set objectStorage.credentials.secretAccessKey="your-secret-access-key" \
+  --set remoteWal.enable=true \
```
Comment (daviderli614) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:95]: After this PR is merged, the installation command will change. It is recommended to try it after merging and then modify the installation command.
https://github.com/GreptimeTeam/greptimedb-operator/pull/180

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 10]
```diff
@@ -1,44 +1,105 @@
...
+  --set remoteWal.kafka.brokerEndpoints=kafka.kafka-cluster.svc.cluster.local:9092 \
+  greptime/greptimedb-cluster \
+  -n greptimedb-cluster
+```
+
+### Resource requests and limits
+To control resource allocation (CPU and memory), modify the Helm installation command as follows:
```
Comment (daviderli614) [inline docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:102]:
```suggestion
To control resource allocation (CPU and Memory), modify the Helm installation command as follows:
```

File: docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md (hunk 8 above)
Comment (WenyXu) [reply docs/user-guide/operations/deploy-on-kubernetes/deploy-greptimedb-cluster.md:59]: Good catch

File: docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 11]
```diff
@@ -1,99 +1,176 @@
...
+  --set objectStorage.s3.region=<region-of-bucket> \
+  --set objectStorage.s3.root=<root-directory-of-data> \
+  --set objectStorage.credentials.accessKeyId=<your-access-key-id> \
+  --set objectStorage.credentials.secretAccessKey=<your-secret-access-key> \
+  greptime/greptimedb-cluster \
+  --create-namespace \
+  -n greptimedb-cluster
+```
+
+### Using RemoteWAL and enable the Region Failover
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:84]:
```suggestion
### Using Remote WAL and enable the Region Failover
```

File: docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 12]
```diff
@@ -1,99 +1,176 @@
...
+If you want to enable RemoteWAL and region failover, follow this configuration. You’ll need a Kafka cluster running, and you can use its endpoint like `kafka.kafka-cluster.svc.cluster.local:9092`:
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:85]:
```suggestion
If you want to enable Remote WAL and region failover, follow this configuration. You’ll need a Kafka cluster running, and you can use its endpoint like `kafka.kafka-cluster.svc.cluster.local:9092`:
```

File: docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 13]
```diff
@@ -1,99 +1,176 @@
...
+
+```bash
+helm install greptimedb \
+  --set meta.etcdEndpoints=etcd.etcd-cluster.svc.cluster.local:2379 \
+  --set meta.enableRegionFailover=true \
+  --set objectStorage.s3.bucket=<your-bucket> \
+  --set objectStorage.s3.region=<region-of-bucket> \
+  --set objectStorage.s3.root=<root-directory-of-data> \
+  --set objectStorage.credentials.accessKeyId=<your-access-key-id> \
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:94]: maybe we can remove the object storage options if they are unnecessary for the remote WAL and region failover?

File: docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 14]
```diff
@@ -1,99 +1,176 @@
...
+For more intricate configurations, first download the default `values.yaml` file and modify it locally.
+
+```bash
+curl -sLo values.yaml https://raw.githubusercontent.com/GreptimeTeam/helm-charts/main/charts/greptimedb-cluster/values.yaml
 ```
 
-Apply this configuration to instantiate the GreptimeDB cluster:
+You can configure each component by specifying custom settings in the `configData` fields. For more details, refer to the [configuration](../configuration.md) documentation. 
+
+Below is an example of how to modify the settings in the `values.yaml` file. This example demonstrates how to configure specific components within GreptimeDB. It sets the metasrv's selector type to `round_robin`, adjusts the datanode’s Mito engine by setting `global_write_buffer_size` to `4GB`, and configures the frontend's meta client `ddl_timeout` to `60s`:
```
Comment (nicecui) [inline docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:144]: The example should use common configurations.
Is these common configurations that users need to be concerned about?

File: docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md (hunk 13 above)
Comment (WenyXu) [reply docs/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:94]: No, the object storage is required for region failover

**Review by daviderli614 — APPROVED** [review]

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 15]
```diff
@@ -21,32 +23,115 @@ helm upgrade \
...
+  --set objectStorage.credentials.accessKeyId=<your-access-key-id> \
+  --set objectStorage.credentials.secretAccessKey=<your-secret-access-key> \
+  --set remoteWal.enable=true \
+  --set remoteWal.kafka.brokerEndpoints[0]=kafka.kafka-cluster.svc.cluster.local:9092 \
+  greptime/greptimedb-cluster \
+  --create-namespace \
+  -n greptimedb-cluster
+```
+
+### 静态身份验证
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:114]:
```suggestion
### 鉴权认证
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 16]
```diff
@@ -21,32 +23,115 @@ helm upgrade \
...
+要为 GreptimeDB 集群启用静态身份验证，您可以在安装期间通过 auth 设置配置用户凭据。以下是示例：
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:115]:
```suggestion
要为 GreptimeDB 集群启用鉴权认证，您可以在安装期间通过 auth 设置配置用户凭据。以下是示例：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 17]
```diff
@@ -21,32 +23,115 @@ helm upgrade \
...
+  --set meta.etcdEndpoints=etcd.etcd-cluster.svc.cluster.local:2379 \
+  --set auth.enabled=true \
+  --set auth.users[0].username=admin \
+  --set auth.users[0].password=admin \
+  greptime/greptimedb-cluster \
+  --create-namespace \
+  -n greptimedb-cluster
+```
+
+### 资源请求与限制
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:130]:
```suggestion
### 资源用量与限制
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 18]
```diff
@@ -56,55 +141,59 @@ helm upgrade \
...
+对于更复杂的配置，首先下载默认的 `values.yaml` 文件并在本地修改。
+
+```bash
+curl -sLo values.yaml https://raw.githubusercontent.com/GreptimeTeam/helm-charts/main/charts/greptimedb-cluster/values.yaml
 ```
 
-使用此配置创建 GreptimeDB 集群：
+您可以通过在 `configData` 字段中指定自定义设置来配置各个组件。更多详细信息，请参阅[配置](../configuration.md)文档。
+
+以下是如何修改 `values.yaml` 文件设置的示例。此示例展示了如何配置 GreptimeDB 中的特定组件。它将 metasrv 的选择器类型设置为 `round_robin`，通过将 datanode 的 Mito 引擎的 `global_write_buffer_size` 设置为 `4GB` 来调整配置，并将 frontend 的 meta 客户端 `ddl_timeout` 设置为 `60s`：
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:158]:
```suggestion
以下是如何修改 `values.yaml` 文件设置的示例。此示例展示了如何配置 GreptimeDB 中的特定组件。它将 metasrv 的 selector 类型设置为 `round_robin`，通过将 datanode 的 Mito 引擎的 `global_write_buffer_size` 设置为 `4GB` 来调整配置，并将 frontend 的 meta 客户端 `ddl_timeout` 设置为 `60s`：
```

File: i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md [hunk 19]
```diff
@@ -56,55 +141,59 @@ helm upgrade \
...
-# gRPC port: 4001
-# MySQL port: 4002
-# PostgreSQL port: 4003
-kubectl port-forward -n greptimedb-cluster svc/greptimedb-frontend 4001:4001 4002:4002 4003:4003 4000:4000 > connections.out &
+# 您可以使用 MySQL 或 PostgreSQL 客户端连接集群，例如： 'mysql -h 127.0.0.1 -P 4002'。
+# HTTP 端口：4000
+# gRPC 端口：4001
+# MySQL 端口：4002
+# PostgreSQL 端口：4003
+kubectl port-forward -n greptimedb-cluster svc/greptimedb-frontend 4000:
```
Comment (nicecui) [inline i18n/zh/docusaurus-plugin-content-docs/current/user-guide/deployments/deploy-on-kubernetes/deploy-greptimedb-cluster.md:196]: Do we need to omit `4000 > connections.out &`?

**Review by nicecui — COMMENTED** [review]
please also update the zh overview.md, thx

**Review by nicecui — APPROVED** [review]

**Pull request merged**